ufl/sorting.py

```diff
@@ -143,9 +143,6 @@ def cmp_expr(a, b):
             # Delve into subtrees
             aops = a.ufl_operands
             bops = b.ufl_operands
-            # Skip subtrees if operands are the same
-            if aops is bops:
-                continue
```

Comment:
Same comment here,  wouldn't this first (removed) check just resolve the comparison quicker?